tool/task.dart

```diff
@@ -391,42 +391,50 @@ Future<String> docPackage({
   var pubPackageDir = Directory.systemTemp.createTempSync(name);
   io_utils.copy(pubPackageDirOrig, pubPackageDir);
 
+  var executable = Platform.executable;
+  var arguments = [
+    '--enable-asserts',
+    path.join(Directory.current.absolute.path, 'bin', 'dartdoc.dart'),
+    '--link-to-remote',
+    '--show-progress',
+    '--show-stats',
+    '--no-validate-links',
+  ];
+  Map<String, String>? environment;
+
   if (pubPackageMetaProvider
       .fromDir(PhysicalResourceProvider.INSTANCE.getFolder(pubPackageDir.path))!
       .requiresFlutter) {
     var flutterRepo =
         await FlutterRepo.fromExistingFlutterRepo(await cleanFlutterRepo);
-    await launcher.runStreamed(flutterRepo.cacheDart, ['pub', 'get'],
-        environment: flutterRepo.env,
-        workingDirectory: pubPackageDir.absolute.path);
-    await launcher.runStreamed(
-      flutterRepo.cacheDart,
-      [
-        '--enable-asserts',
-        path.join(Directory.current.absolute.path, 'bin', 'dartdoc.dart'),
-        '--json',
-        '--link-to-remote',
-        '--show-progress',
-      ],
-      environment: flutterRepo.env,
-      workingDirectory: pubPackageDir.absolute.path,
-      withStats: withStats,
-    );
-  } else {
-    await launcher.runStreamedDartCommand(['pub', 'get'],
-        workingDirectory: pubPackageDir.absolute.path);
-    await launcher.runStreamedDartCommand(
-      [
-        '--enable-asserts',
-        path.join(Directory.current.absolute.path, 'bin', 'dartdoc.dart'),
-        '--json',
-        '--link-to-remote',
-        '--show-progress',
-      ],
-      workingDirectory: pubPackageDir.absolute.path,
-      withStats: withStats,
-    );
+    executable = flutterRepo.cacheDart;
+    environment = flutterRepo.env;
   }
+  await launcher.runStreamed(
+    executable,
+    ['pub', 'get'],
+    environment: environment,
+    workingDirectory: pubPackageDir.absolute.path,
+  );
+  await launcher.runStreamed(
+    executable,
+    // Add the `--json` flag for the support in `runStreamed` to tease out
+    // data and messages.
+    [...arguments, '--json'],
```

Comment:
This is interesting, yeah. What happens when you don't include this flag? I'm having a hard time reasoning with why this is here.

Reply:
I added more to the comment. But briefly, we add `--json` for comparison commands, like comparing the dartdoc output (or warnings) for two different dartdoc versions. And those comparison commands require the stdout to be a structured format, for making the comparison.